src/full/Agda/Syntax/Abstract/Name.hs

```diff
@@ -39,7 +39,8 @@ import Agda.Utils.Impossible
 --   source location of the binding site is also recorded.
 data Name = Name
   { nameId           :: !NameId
-  , nameConcrete     :: C.Name
+  , nameConcrete     :: C.Name  -- ^ The concrete name used for this instance
+  , nameCanonical    :: C.Name  -- ^ The concrete name in the original definition (needed by primShowQName, see #4735)
```

Comment:
Often these two names will be identical.  Should the maybe the `nameCanonical` be a `Maybe C.Name`?
I wonder if there is any impact on serialization time if the name is repeated.

Reply:
Actually, it would be more logical if `nameConcrete` was the `Maybe`.  If `Nothing` it is unchanged from its original, otherwise `Just` carries the new name after the renaming.

Reply:
I don't really see a strong case for making either one a `Maybe`. As argued below I don't think serialization time is an issue, and adding a maybe would just introduce extra logic for looking up the concrete name.

Reply:
Well, equality checking becomes more expensive, since Haskell lacks pointer-equality.